src/client/streaming.rs

```diff
@@ -41,14 +44,19 @@ where T: Message + Default,
         // Get the response
         let response = try_ready!(response);
 
+        let status_code = response.status();
+
         // Destructure into the head / body
         let (head, body) = response.into_parts();
 
-        if let Some(status) = super::check_grpc_status(&head.headers) {
-            return Err(::Error::Grpc(status, head.headers));
+        // Check the headers for `grpc-status`, in which case we should not parse the body.
+        if let Some(status) = infer_grpc_status(&head.headers, None) {
+            if status.code() != Code::OK {
```

Comment:
It'd likely be weird for a `Trailers-Only` response to have `grpc-status = 0`. If the header exists, it's meant to mean a failure and no body... should we assume a status of `OK` to be a mistake?